Automattic/jetpack
# Allow for callables to be retrieved by the sync/object endpoint

This update allows us to expand the checksum process so that we ensure callables are always in sync. It implements get_objects_by_ids and get_object_by_id for the Callables (functions) module that allows for us to retrieve the current value on demand. Pulled callables are limited to the allowed Sync callables.

#### Changes proposed in this Pull Request:

Expand sync/object endpoint to include retrieval of callables.

#### Does this pull request change what data or activity we track or use?


No callables are already retrievable via a Full Sync this simply exposes a new endpoint to request the data.

#### Testing instructions:






1) Access shell on your WP.com sandbox
2) $jsm = new Jetpack_Sync_Manager( get_current_blog_id() );
3) $jsm->get_sync_objects( 'functions', 'callable', array( 'all' ) );
4) Verify that callables are retrieved
5) $jsm->get_sync_objects( 'functions', 'callable', array( 'blue-bird' ) );
6) Verify that empty array is returned
7) Specify your own callable names and that only allowed will be returned.
